lib/ansible/plugins/connection/ssh.py

```diff
@@ -596,10 +602,15 @@ def exec_command(self, *args, **kwargs):
         cmd_summary = "%s..." % args[0]
         for attempt in range(remaining_tries):
             try:
-                return_tuple = self._exec_command(*args, **kwargs)
-                # 0 = success
-                # 1-254 = remote command return code
-                # 255 = failure from the ssh command itself
+                try:
+                    return_tuple = self._exec_command(*args, **kwargs)
+                    # 0 = success
+                    # 1-254 = remote command return code
+                    # 255 = failure from the ssh command itself
+                except (AnsibleInternalError) as e:
+                    # Retry one more time because of the ControlPersist broken pipe
+                    display.vvv(u"RETRYING BECAUSE OF CONTROLPERSIST BROKEN PIPE")
```

Comment:
Ensure that it's the exact right exception (by some mechanism, eg, error code as mentioned above) before we trigger the retry.